cypress-tests/cypress/e2e/PaymentUtils/Xendit.js

```diff
@@ -0,0 +1,624 @@
+        payment_method_data: {
+          card: successfulNo3DSCardDetails,
+          billing: billing_details,
+        },
+        currency: "IDR",
+        mandate_data: singleUseMandateData,
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "succeeded",
+        },
+      },
+    },
+    "3DSManualCapture": {
+      Request: {
+        amount: 6500000,
+        payment_method: "card",
+        payment_method_data: {
+          card: successfulNo3DSCardDetails,
+        },
+        currency: "IDR",
+        customer_acceptance: null,
+        setup_future_usage: "on_session",
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "requires_customer_action",
+          setup_future_usage: "on_session",
+          payment_method_data: payment_method_data_3ds,
+        },
+      },
+    },
+    MandateSingleUseNo3DSManualCapture: {
+      Configs: {
+        DELAY: {
+          STATUS: true,
+          TIMEOUT: 3000,
+        },
+      },
+      Request: {
+        payment_method: "card",
+        amount: 6500000,
+        payment_method_data: {
+          card: successfulNo3DSCardDetails,
+          billing: billing_details,
+        },
+        currency: "IDR",
+        mandate_data: singleUseMandateData,
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "processing",
+        },
+      },
+    },
+    Capture: {
+      Configs: {
+        DELAY: {
+          STATUS: true,
+          TIMEOUT: 3000,
+        },
+      },
+      Request: {
+        payment_method: "card",
+        amount: 6500000,
+        payment_method_data: {
+          card: successfulNo3DSCardDetails,
+        },
+        currency: "IDR",
+        customer_acceptance: null,
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "succeeded",
+          amount: 6500000,
+          amount_capturable: 0,
+          amount_received: 6500000,
+        },
+      },
+    },
+    PartialCapture: {
+      Configs: {
+        DELAY: {
+          STATUS: true,
+          TIMEOUT: 3000,
+        },
+      },
+      Request: {
+        amount: 6500000,
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "partially_captured",
+          amount: 6500000,
+          amount_capturable: 0,
+          amount_received: 100,
+        },
+      },
+    },
+    Refund: {
+      Request: {
+        payment_method: "card",
+        payment_method_data: {
+          card: successfulNo3DSCardDetails,
+        },
+        currency: "IDR",
+        customer_acceptance: null,
+      },
+      Response: {
+        status: 200,
+        body: {
+          status: "succeeded",
+        },
+      },
+    },
+    VoidAfterConfirm: {
+      Request: {},
+      Response: {
+        status: 501,
+        body: {
+          error: {
+            type: "invalid_request",
+            message: "Cancel/Void flow is not supported",
```

Comment:
`501` is not implemented error, right? or, is this expected to be a 4xx?

if this is being thrown from backend, then, i feel, it is wrong, we're supposed to throw a 4xx if not supported.

correct me if wrong.